code/lib/core-server/src/utils/get-story-id.ts

```diff
@@ -0,0 +1,44 @@
+import type { Options } from '@storybook/types';
+import dedent from 'ts-dedent';
+import { normalizeStories, normalizeStoryPath } from '@storybook/core-common';
+import path from 'path';
+import { storyNameFromExport, toId } from '@storybook/csf';
+import slash from 'slash';
+import { userOrAutoTitleFromSpecifier } from '@storybook/preview-api';
+
+interface Data {
+  storyFilePath: string;
+  exportedStoryName: string;
+}
+
+export async function getStoryId(data: Data, options: Options) {
+  const stories = await options.presets.apply('stories', [], options);
+
+  const workingDir = process.cwd();
+
+  const normalizedStories = normalizeStories(stories, {
+    configDir: options.configDir,
+    workingDir,
+  });
+
+  const relativePath = path.relative(workingDir, data.storyFilePath);
+  const importPath = slash(normalizeStoryPath(relativePath));
+
+  const autoTitle = normalizedStories
+    .map((normalizeStory) => userOrAutoTitleFromSpecifier(importPath, normalizeStory))
+    .filter(Boolean)[0];
+
+  if (autoTitle === undefined) {
```

Comment:
```suggestion
  const wasIndexed = normalizedStories
    .some((normalizeStory) => userOrAutoTitleFromSpecifier(importPath, normalizeStory));

  if (!wasIndexed) {
```

Or some other sensible variable name.

Reply:
I need the title later to calculate the story id. It's not just needed to show the error conditionally.

Reply:
I need the `autoTitle` later to create the story id. It is not just needed for the conditional error message.